src/psp/video.c

```diff
@@ -236,6 +236,26 @@ void updateWindowViewport()
         return;
     }
 
+    // For "aspect correct" scaling, fill as much of the window as 4:3 allows
+    if (gScalingMode == ScalingModeAspectCorrect)
+    {
+        if (kWindowWidth * 3 >= kWindowHeight * 4)
+        {
+            gWindowViewport.w = kWindowHeight * 4 / 3;
+            gWindowViewport.h = kWindowHeight;
+            gWindowViewport.x = (kWindowWidth - gWindowViewport.w) / 2;
+            gWindowViewport.y = 0;
+        }
+        else
+        {
+            gWindowViewport.w = kWindowWidth;
+            gWindowViewport.h = kWindowWidth * 3 / 4;
+            gWindowViewport.x = 0;
+            gWindowViewport.y = (kWindowHeight - gWindowViewport.h) / 2;
+        }
+        return;
+    }
+
     float textureAspectRatio = (float)kScreenWidth / kScreenHeight;
```

Comment:
Why not tweaking this `textureAspectRatio` replacing it with `4 / 3` when this mode is enabled, and leave the rest of the logic mainly untouched? Am I missing something?

It applies to the rest of video.c files. I'd also add a comment saying something like `// This should behave as ScalingModeAspectFit with aspect ratio 4:3` for example
```suggestion
    float textureAspectRatio = (float)kScreenWidth / kScreenHeight;
    
    if (gScalingMode == ScalingModeAspect4_3)
    {
        // This should behave as ScalingModeAspectFit with aspect ratio 4:3
        textureAspectRatio = 4.0f / 3.0f;
    }
```

Reply:
I wasn't always sure what's the intention of all modes, other than observing by experimenting. If ScalingModeAspectFit simply implies drawing as much of the contents as possible while preserving a ratio of ~16:10, then I can change the code based on your description, indeed.

Reply:
Yep! That's exactly what it does, right now using the original resolution of the game (320x200). If you replace what I said, it should do the same but with 4:3 instead 😄